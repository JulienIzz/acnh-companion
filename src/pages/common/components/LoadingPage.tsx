import {
  View,
  ActivityIndicator,
  StyleSheet,
  ImageSourcePropType,
} from 'react-native';
import {Header} from '../../header/Header';

type Props = {
  image: ImageSourcePropType;
  text: string;
};

export function LoadingPage({image, text}: Props) {
  return (
    <View
      style={{
        flex: 1,
      }}>
      <Header headerImagePath={image} headerText={text} />
      <View style={styles.loaderPage}>
        <ActivityIndicator />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  loaderPage: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
