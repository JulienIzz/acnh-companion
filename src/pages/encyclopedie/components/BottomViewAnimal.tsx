import React from 'react';
import {SimpleTypography} from '../../../theme/typography/Typography';
import {Animal} from '../Types';

interface BottomViewAnimalProps {
  animal: Animal;
}

export const BottomViewAnimal = ({animal}: BottomViewAnimalProps) => {
  return (
    <>
      <SimpleTypography
        text={'DisponibilitÃ© : ' + animal.availability['month-northern']}
      />
      <SimpleTypography text={'Horaire : ' + animal.availability.time} />
      <SimpleTypography
        text={'Localisation : ' + animal.availability.location}
      />
      {animal.shadow ? (
        <SimpleTypography text={'Ombre : ' + animal.shadow} />
      ) : null}
    </>
  );
};
