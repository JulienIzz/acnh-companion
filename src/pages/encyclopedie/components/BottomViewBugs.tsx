import React from 'react';
import {SimpleTypography} from '../../../theme/typography/Typography';
import {Bug} from '../Types';

interface BottomViewBugsProps {
  animal: Bug;
}

export const BottomViewBugs = ({animal}: BottomViewBugsProps) => {
  return (
    <>
      <SimpleTypography
        text={'DisponibilitÃ© : ' + animal.availability['month-northern']}
      />
      <SimpleTypography text={'Horaire : ' + animal.availability.time} />
      <SimpleTypography
        text={'Localisation : ' + animal.availability.location}
      />
    </>
  );
};
