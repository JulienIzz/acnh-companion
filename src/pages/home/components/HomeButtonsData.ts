import {ImageSourcePropType} from 'react-native/types';
import {Routes} from '../../../navigation/Types';

export interface ButtonData {
  navigationTarget: Routes;
  imagePath: ImageSourcePropType;
  label: string;
}

export const HOME_SCREEN_BUTTONS_DATASET: ButtonData[] = [
  {
    label: 'Encyclop√©die',
    imagePath: require('../img/encyclo_betes.png'),
    navigationTarget: 'Encyclopedie',
  },
  {
    label: 'Recherche',
    imagePath: require('../img/tracker.png'),
    navigationTarget: 'Search',
  },
  {
    label: 'Report de bug',
    imagePath: require('../img/bugs.png'),
    navigationTarget: 'BugReporting',
  },
  {
    label: 'Villageois',
    imagePath: require('../img/villagers.png'),
    navigationTarget: 'Villagers',
  },
];
