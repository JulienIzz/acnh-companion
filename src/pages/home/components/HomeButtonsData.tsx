import {ImageSourcePropType} from 'react-native/types';
import {Routes} from '../../../navigation/Types';

export interface ButtonData {
  navigationTarget: Routes;
  imagePath: ImageSourcePropType;
  label: string;
}

export const HOME_SCREEN_BUTTONS_DATASET: ButtonData[] = [
  {
    label: 'Encyclop√©die',
    imagePath: require('../img/encyclo_betes.png'),
    navigationTarget: 'Encyclopedie',
  },
  {
    label: 'Test',
    imagePath: require('../img/bugs.png'),
    navigationTarget: 'Test',
  },
];
