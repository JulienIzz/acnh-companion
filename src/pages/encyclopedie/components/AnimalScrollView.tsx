import {FlatList, ActivityIndicator} from 'react-native';
import {ExpandableView} from '../../common/components/ExpandableView';
import {Fish, Bug} from '../Types';
import {BottomViewBugs} from './BottomViewBugs';
import {TopViewAnimal} from './TopViewAnimal';
import {BottomViewFish} from './BottomViewFish';
import React from 'react';

type AnimalProps = {type: 'fish'; data: Fish[]} | {type: 'bug'; data: Bug[]};

export const AnimalScrollView = ({type, data}: AnimalProps) => {
  if (type === 'fish') {
    return (
      <FlatList
        horizontal={false}
        data={data}
        renderItem={animal => (
          <ExpandableView
            topView={<TopViewAnimal animal={animal.item} />}
            expandView={<BottomViewFish animal={animal.item} />}
          />
        )}
      />
    );
  }
  if (type === 'bug') {
    return (
      <FlatList
        horizontal={false}
        data={data}
        renderItem={animal => (
          <ExpandableView
            topView={<TopViewAnimal animal={animal.item} />}
            expandView={<BottomViewBugs animal={animal.item} />}
          />
        )}
      />
    );
  } else {
    return <ActivityIndicator />;
  }
};
