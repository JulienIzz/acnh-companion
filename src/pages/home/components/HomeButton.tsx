import {
  StyleSheet,
  View,
  TouchableOpacity,
  Image,
  StyleProp,
  ViewStyle,
} from 'react-native';
import React from 'react';
import {IconTypography} from '../../../theme/typography/Typography';
import {useNavigation} from '@react-navigation/native';
import {ButtonData} from './HomeButtonsData';

type ButtonDataProps = {
  label: ButtonData['label'];
  imagePath: ButtonData['imagePath'];
  navigationTarget: ButtonData['navigationTarget'];
};

type HomeButtonProps = ButtonDataProps & {
  containerStyle?: StyleProp<ViewStyle>;
};

const BUTTON_ICON_HEIGHT_AND_WIDTH = 100;

export const HomeButton = ({
  navigationTarget,
  imagePath,
  label,
  containerStyle,
}: HomeButtonProps) => {
  const navigation = useNavigation();

  return (
    <View style={containerStyle}>
      <TouchableOpacity
        style={styles.touchable}
        onPress={() => navigation.navigate(navigationTarget)}>
        <Image style={styles.image} source={imagePath} />
      </TouchableOpacity>
      <IconTypography text={label} />
    </View>
  );
};

const styles = StyleSheet.create({
  touchable: {
    height: BUTTON_ICON_HEIGHT_AND_WIDTH,
    width: BUTTON_ICON_HEIGHT_AND_WIDTH,
  },
  image: {
    width: '100%',
    height: '100%',
  },
});
