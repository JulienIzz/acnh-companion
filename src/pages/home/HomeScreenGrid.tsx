import {StyleSheet, View, Dimensions} from 'react-native';
import React from 'react';
import {HomeButton, ButtonData} from './components/HomeButton';

export const NUMBER_OF_COLUMNS = 3;
export const MAX_GRID_WIDTH = 500;
export const GRID_WIDTH_RATIO = 0.9;

const BUTTON_CONTAINER_TOTAL_HEIGHT = 160;
const BUTTON_CONTAINER_TOTAL_WIDTH =
  Dimensions.get('window').width > MAX_GRID_WIDTH
    ? (MAX_GRID_WIDTH * GRID_WIDTH_RATIO) / NUMBER_OF_COLUMNS
    : (GRID_WIDTH_RATIO * Dimensions.get('window').width) / NUMBER_OF_COLUMNS;

interface Props {
  buttons: ButtonData[];
}

export const HomeScreenGrid = ({buttons}: Props) => {
  return (
    <View style={styles.listWrapper}>
      {buttons.map(button => (
        <HomeButton
          key={button.label}
          label={button.label}
          imagePath={button.imagePath}
          navigationTarget={button.navigationTarget}
          containerStyle={styles.buttonContainer}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    alignItems: 'center',
    height: BUTTON_CONTAINER_TOTAL_HEIGHT,
    width: BUTTON_CONTAINER_TOTAL_WIDTH,
  },
  listWrapper: {
    alignSelf: 'center',
    marginTop: 40,
    width: `${GRID_WIDTH_RATIO * 100}%`,
    maxWidth: MAX_GRID_WIDTH,
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
});
