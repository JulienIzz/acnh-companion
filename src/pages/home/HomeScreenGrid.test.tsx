import {HomeScreenGrid} from './HomeScreenGrid';
import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {ButtonData} from './components/HomeButtonsData';

jest.mock('@react-navigation/native');

const MOCK_BUTTONS_DATASET: ButtonData[] = [
  {
    label: 'Bouton numéro 1',
    imagePath: require('./img/bugs.png'),
    navigationTarget: 'Test',
  },
  {
    label: 'Bouton numéro 2',
    imagePath: require('./img/bugs.png'),
    navigationTarget: 'Test',
  },
];

test('icon grid with pressable buttons', async () => {
  render(<HomeScreenGrid buttons={MOCK_BUTTONS_DATASET} />);

  const FirstButtonLabel = screen.getByText('Bouton numéro 1');
  const SecondButtonLabel = screen.getByText('Bouton numéro 2');

  expect(FirstButtonLabel).toBeTruthy();
  expect(SecondButtonLabel).toBeTruthy();
  expect(screen.toJSON()).toMatchSnapshot();
});
