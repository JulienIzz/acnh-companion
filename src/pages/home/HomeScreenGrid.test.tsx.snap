// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon grid with pressable buttons 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "marginTop": 40,
      "maxWidth": 500,
      "width": "90%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": 160,
        "width": 150,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": 100,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/pages/home/img/bugs.png",
          }
        }
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
          "textAlign": "center",
        }
      }
    >
      Bouton numéro 1
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "height": 160,
        "width": 150,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": 100,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/pages/home/img/bugs.png",
          }
        }
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
          "textAlign": "center",
        }
      }
    >
      Bouton numéro 2
    </Text>
  </View>
</View>
`;
